### PR TITLE
support reserved resources by guarantee field in QueueSpec doc

### PR DESCRIPTION
**Pull request by qiankunli**
design doc for #1905

### CODE REVIEW

File: docs/design/queue-guarantee-resource-reservation-design.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+1. 只是提交queue6，不提交job
+   a. 此时queue6 没有job，已有代码不会为这个queue6 计算 deserved
+   b. queue6 的guarantee 会参与 已有queue 的deserved 计算，导致已有queue 的 deserved 可能会变小，realCapacity 可能变小，queue 可能变成overused 状态。
+2. 给queue6 提交一个job（queue.guarantee 内）有啥影响？主要是对已经running 的job 的影响
+   a. 集群剩余资源本来就够运行job，影响不大
+   b. 集群剩余资源不够运行job
+
+queue6提交job，集群剩余资源不够运行job
+1. 被overcommit plugin 拦截。job enqueue时，如果集群资源不够运行job 会被overcommit 拦截。此时要更改 overcommit 逻辑，只要job.req < queue6.guarantee，使pg 从pending 进入enqueue 状态，
```
Comment (william-wang) [inline docs/design/queue-guarantee-resource-reservation-design.md:90]: please change "enqueue" to "inqueue" status. enqueue is a action of volcano. "inqueue" is a phase of podgroup.

File: docs/design/queue-guarantee-resource-reservation-design.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+2. 之后queue6 job 的task 处于pending 状态，触发reclaim 工作，找到overused queue 删除task（假设删除的是queue1 的task）
+3. 删除queue1 task 可能会导致queue1 的某个job 从running 变为 饥饿状态(readyTaskNum < minAvailable)，且queue1 处于overused 状态。 这类job 会浪费资源
+4. 为此需要清理 overused 状态下 饥饿 的job，这段逻辑
```
Comment (william-wang) [inline docs/design/queue-guarantee-resource-reservation-design.md:93]: A better way to let the preemptee job release resouce is to configure event/action for job. Make job completed when it's task is evicted. No need to write a new action.

File: docs/design/queue-guarantee-resource-reservation-design.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+1. queue 配置没有变化时 保证guarantee 机制
+2. 新增一个带guarantee queue 或某个queue 的guarantee 变大时保证guarantee 机制
+
+### queue guarantee 配置没有变化时 保证guarantee 机制
+
+每一次开始调度之前，volcano 会为每个queue 计算好deserved 资源（代表了queue 本轮调度可以使用的资源），queue 中的task 要运行时，如果deserved > allocated, task 即可运行。为保证 guarantee 机制
+
+1. 任何时候，queue 的deserved 应大于 queue.guarantee。
+2. 如果有一个queue 设置了guarantee，相当于剩下的所有的queue 都设置了一个上限值，我们称之为realCapacity
+   a. 如果一个queue 未设置capacity，`realCapacity = 资源总量 - 其它queue的guarantee 之和`
```
Comment (william-wang) [inline docs/design/queue-guarantee-resource-reservation-design.md:59]: the "realCapaciy" a confusing word, let's get another name. like "limit" or some other words.

**Comment by Thor-wl** [comment]
Pls keep the doc all in English. THX.

**Comment by hwdef** [comment]
I think the documentation needs to be more detailed. 
For example, the usage scenarios of this feature, why do we need this feature.

**Comment by qiankunli** [comment]
> I think the documentation needs to be more detailed. For example, the usage scenarios of this feature, why do we need this feature.

ok, added

File: docs/design/queue-guarantee-resource-reservation-design.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
+# Volcano Resource Reservation For Queue
+
+@[qiankunli](https://github.com/qiankunli); Oct 11rd, 2021
+
+## Motivation
+
+In my case, we use volcano to schedule training job(tfjob/pytorchjob/vcjob) in k8s cluster, there are many groups such as ad/recommend/tts, a queue represents a group.
+In order to ensure the full utilization of resources, we generally do not configure `queue.capability`. but this will cause a queue to running out all resources, and make the new job of other queue unable to execute.
+so we want to reserve some resources for a queue, so that any new job in the queue can be submitted immediately.
```
Comment (hwdef) [inline docs/design/queue-guarantee-resource-reservation-design.md:9]: If I only have one queue. Will this feature play any role?

**Review by william-wang — APPROVED** [review]
/lgtm

File: docs/design/queue-guarantee-resource-reservation-design.md (hunk 4 above)
Comment (qiankunli) [reply docs/design/queue-guarantee-resource-reservation-design.md:9]: the queue can running out all the resources in k8s cluster

**Pull request merged**